app/models/spree_adyen/gateway/payment_setup_sessions.rb

```
@@ -0,0 +1,159 @@
+          client.checkout.payments_api.payments_details({ details: { redirectResult: redirect_result } })
+        end
+        result_code = response.response&.dig('resultCode')
+        status = case result_code
+                 when 'Authorised' then 'completed'
+                 when 'Pending', 'Received' then 'paymentPending'
+                 when 'Cancelled' then 'canceled'
+                 when 'Refused', 'Error' then 'refused'
+                 else result_code
+                 end
+        process_setup_session_status(setup_session, status, response.response || {})
+      end
```

RuboCop (1.86.1)

[convention] Method has too many lines. [12/10]

(Metrics/MethodLength)